runtime/lua/vim/lsp.lua

```diff
@@ -474,10 +480,22 @@ local function validate_cmd(v)
   return type(v) == 'function'
 end
 
+--- Augments a validator function with support for optional (false) values.
+--- @param f string
+--- @return fun(v): boolean # The augmented function. Also returns true if {v} is
+local function optional_validator(f)
+  return function(v)
+    if type(f) == 'string' then
```

Comment:
Isn't `f` always a string?

Reply:
Yes, copy and paste leftover from client.lua.